### ISSUE TITLE
Networking benchmark use internal IP to ping, so VMs on different subnets are not reachable

### ISSUE THREAD

**Issue by arjunaugustine** [GoogleCloudPlatform/PerfKitBenchmarker#1056]
I used a combination of one personal static server and one google cloud VM to try run ping benchmark. I found that the benchmark was using internal IPs in its run phase, so both the machines were unable to ping each other, since they were on different networks. To try solve the issue, I provided the external IP of the static_vm in place of the internal IP, and now the google VM was able to ping my server, but not vice versa. I also tried providing "--ip_addresses=EXTERNAL" in the command line used to execute the benchmark, to no results. Kindly provide some insights on how to solve this issue!

Warning message: <Google VM external IP> is not reachable from <Static VM External IP>
Here is the static_vm_file that I used:
```
[
  {
    "ip_address": "<External IP of my machine>",
    "user_name": "root",
    "keyfile_path": "/root/.ssh/id_rsa",
    "internal_ip": "<External IP of my machine>"
  }
]
```
And this is the command line option that I used:
`./pkb.py --benchmarks=ping --static_vm_file=india.json --ip_addresses=EXTERNAL`

**Comment by voellm** [comment]
This seems like the Google VM has a firewall rule set.  By default Google Cloud block ping.

Can you check the firewall?

**Comment by arjunaugustine** [comment]
Yes, this was a firewall issue that came up in providers/gcp/gce_network.py:
When assigning self.default_firewall_rule, the script tries to allow connections only from a network range : '10.0.0.0/16'. This means that when using 2 sets of internal networks (In my case, GCP and my internal server) the accesses were coming in from an external IP and this was getting blocked by the default firewall rule. I changed the rule to remove the network range restriction in its entirety. I couldn't use a network range that included the external IP of my server since the range has to come under RFC1918 private address blocks (10.0.0.0/16, 172.16.0.0/12, 192.168.0.0/16). Also for running other benchmarks like iperf in a similar setup, I had to edit the firewall_rule assignment line to 'ALLOW_ALL' ports to connect to the network instead of just a subset.
These changes are not required for running benchmarks in a GCP only mode, so I am not sure if they should come out as a standard change, considering security issues.
